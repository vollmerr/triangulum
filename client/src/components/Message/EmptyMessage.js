import React from 'react';
import { Button, Header } from 'semantic-ui-react';

const EmptyMessage = ({ message }) => {
  const centerProps = {
    style: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      textAlign: 'center',
      minHeight: '100vh',
      marginTop: '-10vh',
    },
  };

  const messageProps = {
    as: 'h3',
    textAlign: 'center',
    color: 'violet',
    content: message,
    style: { padding: '20px 0' },
  };

  const buttonProps = {
    icon: 'plus',
    type: 'button',
    color: 'violet',
    content: 'New Crawl',
    onClick: () => { window.location.hash = '/'; },
  };

  return (
    <div {...centerProps}>
      <Header {...messageProps} />
      <Button {...buttonProps} />
    </div>
  );
};

export default EmptyMessage;
