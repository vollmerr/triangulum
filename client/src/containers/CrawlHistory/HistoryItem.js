import React from 'react';
import { List } from 'semantic-ui-react';

const HistoryItem = item => (
  <List.Item>
    <List.Content floated='right'>
      <List.Description>{new Date(item.updated).toUTCString()}</List.Description>
    </List.Content>
    <List.Content>
      <List.Header>{item.request.url}</List.Header>
      <List.Description>{`Type: ${item.request.type}`}</List.Description>
      <List.Description>{`Limit: ${item.request.limit}`}</List.Description>
      {
        item.request.keyword
        && <List.Description>{`Keyword: ${item.request.keyword}`}</List.Description>
      }
    </List.Content>
  </List.Item>
);

export default HistoryItem;
