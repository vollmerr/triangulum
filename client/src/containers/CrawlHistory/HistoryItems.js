import React from 'react';
import { List } from 'semantic-ui-react';

import EmptyMessage from '../../components/Message/EmptyMessage';
import HistoryItem from './HistoryItem';

const HistoryItems = ({ items, onClick }) => {
  const keys = Object.keys(items);

  // No history message
  if (!keys.length) {
    return (
      <EmptyMessage message={'No History :('} />
    );
  }

  return (
    <List selection divided verticalAlign='middle'>
      {
        keys.map(x => (
          <HistoryItem key={x} {...items[x]} onClick={onClick} />
        ))
      }
    </List>
  );
};

export default HistoryItems;
