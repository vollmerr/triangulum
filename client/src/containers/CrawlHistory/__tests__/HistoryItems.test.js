import React from 'react';
import { shallow } from 'enzyme';
import { Header } from 'semantic-ui-react';

import EmptyMessage from '../../../components/Message/EmptyMessage';
import HistoryItem from '../HistoryItem';
import HistoryItems from '../HistoryItems';

const props = {
  items: {
    id1: { k1: 'v1', k2: 'v2' },
    id2: { k3: 'v3', k4: 'v4' },
  },
};

let wrapper;
describe('HistoryItem', () => {
  beforeEach(() => {
    wrapper = shallow(<HistoryItems {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a `HistoryItem` for each item', () => {
    const numItems = Object.keys(props.items).length;
    expect(wrapper.find(HistoryItem).length).toEqual(numItems);
  });

  it('should render a message if no history', () => {
    wrapper.setProps({ items: {} });
    expect(wrapper.find(EmptyMessage).length).toEqual(1);
  });
});
