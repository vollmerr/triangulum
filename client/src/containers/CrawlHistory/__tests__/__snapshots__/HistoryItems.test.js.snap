// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem should render correctly 1`] = `
<List
  divided={true}
  selection={true}
  verticalAlign="middle"
>
  <HistoryItem
    k1="v1"
    k2="v2"
    key="id1"
  />
  <HistoryItem
    k3="v3"
    k4="v4"
    key="id2"
  />
</List>
`;
